### PR TITLE
Include Jupyter notebooks in Sphinx documentation using myst-nb

### PR DESCRIPTION
**Pull request by seisman**
Changes in this PR:

- Include notebooks using myst-nb
- Install dependencies using conda, instead of pip (because we need obspy)
- Pin Python to 3.9 and obspy to 1.2.2 so that we have stable notebook outputs
- Remove "requirements.txt" and add "environment.yml"
- Cache notebook outputs in GitHub Actions
- Run the "deploy.yml" workflow weekly, and force the re-execution of all notebooks

### CODE REVIEW

**Comment by seisman** [comment]
@core-man I think this PR is ready for review now. Please see the PR description for changes in this PR.

As a PR should focus on one small thing, I plan to remove the `distaz.md` file from this PR before merging.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -8,24 +8,40 @@ on:
...
+  schedule:
+    # Run weekly at 00:00 on Sunday
+    - cron: '0 0 * * 0'
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
```
Comment (core-man) [inline .github/workflows/deploy.yml:20]: I don't find the meaning of `-l` and `{0}`. Could you show me some links?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -8,24 +8,40 @@ on:
...
+          path: build/.jupyter_cache
+          key: notebooks
 
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
+      - name: Disable notebook cache for scheduled jobs
+        if: github.event_name == 'schedule'
+        run: sed -i"" 's/jupyter_execute_notebooks = "cache"/jupyter_execute_notebooks = "force"/' source/conf.py
```
Comment (core-man) [inline .github/workflows/deploy.yml:44]: `-i""` or `-i`?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (seisman) [reply .github/workflows/deploy.yml:20]: 1. First point in https://github.com/conda-incubator/setup-miniconda#important
2. https://github.com/conda-incubator/setup-miniconda#use-a-default-shell

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (seisman) [reply .github/workflows/deploy.yml:44]: `-i""` works for both GNU sed and BSD sed, while `-i` only works for GNU seed.

File: source/exercises/distaz.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.0
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
```
Comment (core-man) [inline source/exercises/distaz.md:13]:
```suggestion

This notebook can be downloaded as
            **{nb-download}`distaz.ipynb`** and {download}`distaz.md`.

```

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: seismology101
+channels:
+    - conda-forge
+    - nodefaults
+dependencies:
+    - python=3.9
+    - pip
+    - sphinx
+    - sphinx-copybutton
+    - sphinx-panels
```
Comment (core-man) [inline environment.yml:10]:
```suggestion
    - sphinx_design
```

File: environment.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    - python=3.9
+    - pip
+    - sphinx
+    - sphinx-copybutton
+    - sphinx_design
+    - sphinx_rtd_theme
+    - myst-nb
+    - obspy=1.2.2
+    - pip:
+      - sphinx-cjkspace
```
Comment (core-man) [inline environment.yml:15]:
```suggestion
    - sphinx_rtd_theme
    - myst-nb
    - obspy=1.2.2
    - pip:
      - sphinx-cjkspace
      - sphinx_design
```

File: source/exercises/distaz.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+This notebook can be downloaded as
+            **{nb-download}`distaz.ipynb`** and {download}`distaz.md`.
```
Comment (core-man) [inline source/exercises/distaz.md:15]:
```suggestion
This notebook can be downloaded as
{nb-download}`distaz.ipynb` and {download}`distaz.md`.
```

**Review by core-man — APPROVED** [review]

**Pull request merged**